include/flux/op/cartesian_base.hpp

```diff
@@ -222,7 +222,9 @@ struct cartesian_traits_base_impl {
                   && (sized_sequence<Bases> && ...))
     {
         return std::apply([](auto&... base) {
-            return (flux::size(base) * ...);
+            distance_t sz = 1;
+            ((sz = flux::num::checked_mul(sz, flux::size(base))), ...);
```

Comment:
Is `flux::num` needed instead of just `num::`? It seems most places in the codebase use just `num::`.

Reply:
Yep, well spotted, this could just be `num::checked_mul`